### DISCUSSION TITLE
DTO x ViewModel

### DISCUSSION THREAD

**Discussion by Lippi99** [balta-io/forum#669] [Tira Dúvidas] [answered]
Olá. Conforme fui avançando no curso, me deparei com o conceito de ViewModel, que é uma classe que utilizamos como dado de entrada ou dado de saída, que seria por exemplo o cadastro de uma informação baseado em uma model e a resposta do servidor a partir de uma model também.

Porém, ao pesquisar na internet, me deparei com o conceito de DTO (Data Transfer Object), que pelo que entendi, tem o objetivo de retornar um dado de saída a partir de uma model também, mas sem a responsabilidade de fazer validações como geralmente fazemos com o DataAnnotation nas classes das ViewModels.

Sei que no final disso tudo, tanto DTO como ViewModel são classes e não faria diferença no resultado. Mas qual seria a diferença de fato entre ambas?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @Lippi99, veja se isto te ajuda:
https://github.com/balta-io/forum/discussions/655
